cmd/minikube/cmd/docker-env.go

```diff
@@ -193,9 +199,37 @@ func mustRestartDockerd(name string, runner command.Runner) {
 		// if we get to the point that we have to restart docker (instead of reload)
 		// will need to wait for apisever container to come up, this usually takes 5 seconds
 		// verifying apisever using kverify would add code complexity for a rare case.
-		klog.Warningf("waiting 5 seconds to ensure apisever container is up...")
-		time.Sleep(time.Second * 5)
+		klog.Warningf("waiting to ensure apisever container is up...")
+		startTime := time.Now()
+		if err = WaitForAPIServerProcess(runner, startTime, time.Second*30); err != nil {
+			exit.Message(reason.RuntimeRestart, `The api server within '{{.name}}' is not up`, out.V{"name": name})
```

Comment:
since apiserver is not essential to Docker Env command, we should not exit on this line. I think a klogwWarning is enough.